pkg/controller/bootstrap/jws.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+import (
+	"fmt"
+	"strings"
+
+	jose "github.com/square/go-jose"
+)
+
+func computeDetachedSig(content, tokenID, tokenSecret string) (string, error) {
```

Comment:
The name is a bit odd... I'd find something like `computeDetachedJWSSignature()` clearer, i.e. it's implied that "detached" is well-known thing in JWS terminology. Or even something simpler like `signWithJWS()`.